File: DCC/Distributed File System.py
import math

def main():
    N, F = map(int, input().split())
    current_node = 1
    for _ in range(F):
        file_info = input().split()
        file_name = file_info[0]
        file_size = int(file_info[1])
        replication_factor = int(file_info[2])
        num_chunks = math.ceil(file_size / 64)
        print(f"File {file_name}:")
        for chunk in range(1, num_chunks + 1):
            nodes = []
            for _ in range(replication_factor):
                nodes.append(f"Node {current_node}")
                current_node += 1
                if current_node > N:
                    current_node = 1
            print(f"Chunk {chunk}: " + " ".join(nodes))

if __name__ == "__main__":
    main()

File: DCC/Distributed File System.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int main() {
    int N, F;
    cin >> N >> F;

    int currentNode = 1;

    for (int i = 0; i < F; i++) {
        string fileName;
        int fileSize, replicationFactor;
        cin >> fileName >> fileSize >> replicationFactor;

        int numChunks = ceil(static_cast<double>(fileSize) / 64);
        cout << "File " << fileName << ":" << endl;

        for (int chunk = 1; chunk <= numChunks; chunk++) {
            vector<string> nodes;
            
            for (int j = 0; j < replicationFactor; j++) {
                nodes.push_back("Node " + to_string(currentNode));
                currentNode++;
                if (currentNode > N) {
                    currentNode = 1;
                }
            }

            cout << "Chunk " << chunk << ": ";
            for (int j = 0; j < nodes.size(); j++) {
                cout << nodes[j];
                if (j < nodes.size() - 1) {
                    cout << " ";
                }
            }
            cout << endl;
        }
    }

    return 0;
}

